ush/python/pygfs/jedi/jedi.py

```diff
@@ -152,119 +184,150 @@ def link_exe(task_config: AttrDict) -> None:
+    def extract_tar_from_filehandler_dict(filehandler_dict) -> None:
+        """Extract tarballs from FileHandler input dictionary
 
-        This method extract files from a tarball
+        This method extracts files from tarballs specified in a FileHander
+        input dictionary for the 'copy' action.
 
         Parameters
         ----------
-        tar_file
-            path/name of tarball
+        filehandler_dict
+            Input dictionary for FileHandler
 
         Returns
         ----------
         None
         """
 
-        # extract files from tar file
-        tar_path = os.path.dirname(tar_file)
-        try:
-            with tarfile.open(tar_file, "r") as tarball:
-                tarball.extractall(path=tar_path)
-                logger.info(f"Extract {tarball.getnames()}")
-        except tarfile.ReadError as err:
-            if tarfile.is_tarfile(tar_file):
-                logger.error(f"FATAL ERROR: {tar_file} could not be read")
-                raise tarfile.ReadError(f"FATAL ERROR: unable to read {tar_file}")
+        for item in filehandler_dict['copy']:
+            # Use the filename from the destination entry if it's a file path
+            # Otherwise, it's a directory, so use the source entry filename
+            if os.path.isfile(item[1]):
+                filename = os.path.basename(item[1])
             else:
-                logger.info()
-        except tarfile.ExtractError as err:
-            logger.exception(f"FATAL ERROR: unable to extract from {tar_file}")
-            raise tarfile.ExtractError("FATAL ERROR: unable to extract from {tar_file}")
+                filename = os.path.basename(item[0])
+
+            # Check if file is a tar ball
+            if os.path.splitext(filename)[1] == '.tar':
+                tar_file = f"{os.path.dirname(item[1])}/{filename}"
+
+                # Extract tarball
+                logger.info(f"Extract files from {tar_file}")
+                extract_tar(tar_file)
+
+
+@logit(logger)
+def extract_tar(tar_file: str) -> None:
+    """Extract files from a tarball
+
+    This method extract files from a tarball
+
+    Parameters
+    ----------
+    tar_file
+        path/name of tarball
+
+    Returns
+    ----------
+    None
+    """
+
+    # extract files from tar file
+    tar_path = os.path.dirname(tar_file)
+    try:
+        with tarfile.open(tar_file, "r") as tarball:
+            tarball.extractall(path=tar_path)
+            logger.info(f"Extract {tarball.getnames()}")
+    except tarfile.FileExistsError as err:
+        logger.exception(f"FATAL ERROR: {tar_file} does not exist")
+        raise tarfile.FileExistsError(f"FATAL ERROR: {tar_file} does not exist")
+    except tarfile.ReadError as err:
+        if tarfile.is_tarfile(tar_file):
+            logger.error(f"FATAL ERROR: {tar_file} could not be read")
+            raise tarfile.ReadError(f"FATAL ERROR: unable to read {tar_file}")
+        else:
+            logger.info()
```

Comment:
What is the benefit/use of this `else` block?

Reply:
@RussTreadon-NOAA Do you know what was supposed to go in this else block? I think this code dates back to [#2875](https://github.com/NOAA-EMC/global-workflow/pull/2875)

Reply:
I was avail to get the jist of what was supposed to go here and filled it in in my latest commit